src/ImageSharp/Formats/Jpeg/Components/RowOctet.cs

```diff
@@ -56,13 +41,56 @@ public RowOctet(Buffer2D<T> buffer, int startY)
                     7 => this.row7,
                     _ => ThrowIndexOutOfRangeException()
                 };
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            private set
+            {
+                switch (y)
+                {
+                    case 0:
+                        this.row0 = value;
+                        break;
+                    case 1:
+                        this.row1 = value;
+                        break;
+                    case 2:
+                        this.row2 = value;
+                        break;
+                    case 3:
+                        this.row3 = value;
+                        break;
+                    case 4:
+                        this.row4 = value;
+                        break;
+                    case 5:
+                        this.row5 = value;
+                        break;
+                    case 6:
+                        this.row6 = value;
+                        break;
+                    default:
+                        this.row7 = value;
+                        break;
+                }
             }
         }
 
-        [MethodImpl(InliningOptions.ColdPath)]
-        private static Span<T> ThrowIndexOutOfRangeException()
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void Update(Buffer2D<T> buffer, int startY)
         {
-            throw new IndexOutOfRangeException();
+            int y = startY;
+            int height = buffer.Height;
+
+            // We don't actually have to assign values outside of the
+            // frame pixel buffer since they are never requested.
+            for (int i = 0; i < 8 && y < height; i++)
```

Comment:
That's not quite the same since height is the frame pixel buffer height. You'd end up always doing 8 runs even at the end of the image. 

You did prompt me to rewrite it using a while loop precalculating the yEnd value first.

Reply:
Yeah missed that these are two different variables.